dali/pipeline/operators/decoder/nvjpeg_decoder_new.h

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_DECODER_NEW_H_
+#define DALI_PIPELINE_OPERATORS_DECODER_NVJPEG_DECODER_NEW_H_
+
+#include <nvjpeg.h>
+#include <functional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/operators/decoder/nvjpeg_helper.h"
+#include "dali/util/image.h"
+#include "dali/util/ocv.h"
+#include "dali/image/image_factory.h"
+#include "dali/pipeline/util/thread_pool.h"
+
+namespace dali {
+
+class nvJPEGDecoderNew : public Operator<MixedBackend> {
+ public:
+  explicit nvJPEGDecoderNew(const OpSpec& spec) :
+    Operator<MixedBackend>(spec),
+    output_image_type_(spec.GetArgument<DALIImageType>("output_type")),
+    hybrid_huffman_threshold_(spec.GetArgument<unsigned int>("hybrid_huffman_threshold")),
+    output_info_(batch_size_),
+    image_decoders_(batch_size_),
+    image_states_(batch_size_),
+    jpeg_streams_(batch_size_),
+    decode_params_(batch_size_),
+    pinned_buffer_(batch_size_),
+    decoder_host_state_(batch_size_),
+    decoder_huff_hybrid_state_(batch_size_),
+    output_shape_(batch_size_),
+    device_buffer_(num_threads_),
+    streams_(num_threads_),
+    events_(num_threads_),
+    thread_pool_(num_threads_,
+                 spec.GetArgument<int>("device_id"),
+                 true /* pin threads */) {
+    NVJPEG_CALL(nvjpegCreateSimple(&handle_));
+
+    size_t device_memory_padding = spec.GetArgument<Index>("device_memory_padding");
+    size_t host_memory_padding = spec.GetArgument<Index>("host_memory_padding");
+    NVJPEG_CALL(nvjpegSetDeviceMemoryPadding(device_memory_padding, handle_));
+    NVJPEG_CALL(nvjpegSetPinnedMemoryPadding(host_memory_padding, handle_));
+
+    // to create also in GPU Op
+    NVJPEG_CALL(nvjpegDecoderCreate(
+                      handle_, NVJPEG_BACKEND_HYBRID, &decoder_huff_host_));
+    NVJPEG_CALL(nvjpegDecoderCreate(
+                      handle_, NVJPEG_BACKEND_GPU_HYBRID, &decoder_huff_hybrid_));
+
+
+    for (int i = 0; i < batch_size_; i++) {
+      NVJPEG_CALL(nvjpegDecodeParamsCreate(handle_, &decode_params_[i]));
+      NVJPEG_CALL(nvjpegDecodeParamsSetOutputFormat(decode_params_[i],
+                                                    GetFormat(output_image_type_)));
+
```

Comment:
CMYK images can be processed to match opencv output. Add following call to nable that: 
nvjpegDecodeParamsSetAllowCMYK(decode_params, 1);

Reply:
Done